pkg/user/tx_client_test.go

```diff
@@ -447,3 +452,241 @@ func setupEstimatorService(t *testing.T) *mockEstimatorServer {
+
+			primaryConn := conns[0]
+			otherConns := conns[1:]
+
+			// Create a temporary TxClient for this test case using the suite's signer/config
+			// but the mock connections.
+			tempTxClient, err := user.NewTxClient(
+				suite.encCfg.Codec,
+				suite.txClient.Signer(), // Reuse signer from main client
+				primaryConn,             // Use mock primary connection
+				suite.encCfg.InterfaceRegistry,
+				user.WithAdditionalCoreEndpoints(otherConns), // Use mock secondaries
+				user.WithDefaultAccount(suite.txClient.DefaultAccountName()), // Use main client's default account
+			)
+			require.NoError(t, err, "Failed to create temporary TxClient for test case %d", i)
+
+			var ctx context.Context
+			var cancel context.CancelFunc
+			if errors.Is(tc.expectedErr, context.DeadlineExceeded) {
+				ctx, cancel = context.WithTimeout(context.Background(), 50*time.Millisecond) // Short timeout for deadline test
+			} else {
+				ctx, cancel = context.WithTimeout(context.Background(), 10*time.Second) // General timeout
+			}
+			defer cancel()
+
+			resp, err := tempTxClient.BroadcastTx(ctx, []sdk.Msg{defaultMsg}, defaultOpts...)
+
+			if tc.expectedErr == nil {
+				require.NoError(t, err, "Expected success, but got error: %v", err)
+				require.NotNil(t, resp, "Expected non-nil response on success")
+				require.Equal(t, abci.CodeTypeOK, resp.Code, "Expected CodeTypeOK on success")
+			} else {
+				require.Error(t, err, "Expected error '%v', but got nil error", tc.expectedErr)
+
+				isCorrectError := false
+				var broadcastErr *user.BroadcastTxError
+
+				// Check for specific BroadcastTxError type first
+				if errors.As(err, &broadcastErr) {
+					if _, ok := tc.expectedErr.(*user.BroadcastTxError); ok {
+						isCorrectError = true
+					}
+				} else {
+					switch {
+					case errors.Is(tc.expectedErr, context.DeadlineExceeded):
+						// Check actual error is DeadlineExceeded (could be wrapped or gRPC status)
+						if errors.Is(err, context.DeadlineExceeded) || isGrpcDeadlineError(err) {
+							isCorrectError = true
+						}
+					case errors.Is(tc.expectedErr, errMock1):
+						// Special check for "All Fail" case - any mock error is acceptable
+						if errors.Is(err, errMock1) || errors.Is(err, errMock2) || errors.Is(err, errMock3) {
+							isCorrectError = true
+						} else {
+							// Fallback check: Check if the error string contains any mock error message
+							errStr := err.Error()
+							if strings.Contains(errStr, errMock1.Error()) || strings.Contains(errStr, errMock2.Error()) || strings.Contains(errStr, errMock3.Error()) {
+								isCorrectError = true
+							}
+						}
+					default:
+						if errors.Is(err, tc.expectedErr) {
+							isCorrectError = true
+						}
+					}
+				}
+
+				require.True(t, isCorrectError, "Error mismatch. Expected error matching type/value '%T(%v)', but received '%T(%v)'", tc.expectedErr, tc.expectedErr, err, err)
+			}
+		})
+	}
+}
+
+func isGrpcDeadlineError(err error) bool {
```

Comment:
is this actually necessary? Does the error get wrapped? Regardless, I believe errors.Is checks target error for equivalency (unwrapping). Not sure you actually need this.

Reply:
errors.Is didn't work for me 
will double check the wrapping options